### PR TITLE
[draft] docs: add 3.3.0 relese notes

### PR DESCRIPTION
**Pull request by janl**

### CODE REVIEW

File: src/docs/src/whatsnew/3.3.rst [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+Version 3.3.0
+=============
+
+Features and Enhancements
+-------------------------
+
+* :ghissue:`3746`: Remove couch_icu_driver module and re-use ``couch_ejson_compare`` functionality.
+
+* :ghissue:`3776`, :ghissue:`3970`, :ghissue:`3972`, :ghissue:`4093`, :ghissue:`4102`, :ghissue:`4104`, :ghissue:`4110`, :ghissue:`4111`, :ghissue:`4114`, :ghissue:`4245`, :ghissue:`4246`: Add ``smoosh`` queue persistence. This allows resuming ``smoosh`` operations after a node restart. This is disabled by default and can be enabled with ``[smoosh] persist = true``.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:33]: #3776 is "Bump jiffy to CouchDB-1.0.9-1"

File: src/docs/src/whatsnew/3.3.rst [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`3853`: Remove legacy ``ddoc_cache_opener`` gen_server and speed up event routing.
+
+* :ghissue:`3879`: Remove use of ``ERL_OPTS`` environment variable. All supported Erlang version now use ``ERL_COMPILER_OPTIONS`` for the same purpose.
+
+* :ghissue:`3883`: Add support for SpiderMonkey 91.
+
+* :ghissue:`3889`: Track ``libicu`` collator versions in the view header
+
+* :ghissue:`3925`: Make the timeout for receiving requests from attachment writers configurable.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:53]: #3925 is: Fix publication of nightly packages

File: src/docs/src/whatsnew/3.3.rst [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`3927`: Include index signature in ``_search_info``.
+
+* :ghissue:`3963`: Optimtize key tree stemming by using maps instead of sets. This greatly reduced memory usage for heavily conflicted docs in some situations.
+
+* :ghissue:`3974`: Create new config options in ``[couchdb]`` and ``[smoosh]`` sections to enable finer control of compaction logging levels.
+
+* :ghissue:`3983`, :ghissue:`3984`, :ghissue:`3985`, :ghissue:`3987` :ghissue:`#4033`: Add various functions to ``couch_debug`` module.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:61]: remove # in 4033

File: src/docs/src/whatsnew/3.3.rst [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`4000`: Ensure ``Object.prototype.toSource()`` is always available
+
+* :ghissue:`4017`: Update ``mochiweb`` to 3.1.1.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:65]: #4017 is: Bump mochiweb to 3.0

File: src/docs/src/whatsnew/3.3.rst [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`4140`, :ghissue:`4162`: Upgrade hash algorithm for cookie auth (sha1 -> sha256). This introduces a new config setting ``hash_algorithms``. New cookie values are hashed with sha256, sha1 hashes are still accepted. Admins can set this to sha256 only. Sha1 will be disallowed in the next major release. Show supported hash algorithms in ``_versions`` endpoint.
+
+* :ghissue:`4179`: Don't double-encode changes sequence strings in the replicator.
+
+* :ghissue:`4182`: Explicitly maintain a fully connected cluster. Previously, it was possible for the nodes to disconnect, and for that state to persist util the nodes restarted.
+
+* :ghissue:`4177`: Redact passwords in log file.
+
+* :ghissue:`4259`: The ``_dbs_info`` access control is now configured with the ``[couchdb] admin_only_all_dbs`` setting. Defaults to true. This was a leftover from the 3.0.0 release.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:95]: 4259 doesn't exists. #4254?

File: src/docs/src/whatsnew/3.3.rst [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`4020`: Fix ``maybe_handle_error`` clauses.
+
+* :ghissue:`4037`: Fix ES{256,384,512} support for JWTs.
+
+* :ghissue:`4040`: Handle ``exit(shutdown)`` error in ``chttpd``.
+
+* :ghissue:`4043`: Fix purge request timeouts (5s -> infinity).
+
+* :ghissue:`4046`: The ``devcontainer`` has been updated.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:175]: #4046 is:  Add smoosh state diagram to directory and reference it.

#4146: Update devcontainer

File: src/docs/src/whatsnew/3.3.rst [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`4050`: Handle ``all_dbs_active`` in ``fabric_doc_update``.
+
+* :ghissue:`4061`: Return a proper ``400`` error when an invlaid object is sent to ``_bulk_get``.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:179]: #4061 is: Skip PPC64 CI builds

#2726 Return a 400 error code for an invalid update sequence - but it is from 03/2020 ...

File: src/docs/src/whatsnew/3.3.rst [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`3830`: Reduce skipped Elixir integration tests.
+
+* :ghissue:`3890`: Handle ``not_found`` lookups removing ddoc cache key
+
+* :ghissue:`3892`: Use Debian Stable for CI, add Erlang 24 to CI.
+
+* :ghissue:`3898`: Remove CI support for Ubuntu 16.04.
+
+* :ghissue:`3903`:, :ghissue:`3914`: Refactor Jenkins to dynamically generate stages. Drop ``MINIMUM_ERLANG_VERSION`` to 20, drop the packaging ``ERLANG_VERSION`` to 23, aAd the ``weatherreport-test`` as a build step, and add ``ARM`` and ``POWER`` back into the matrix.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:214]: typo aAd -> add

File: src/docs/src/whatsnew/3.3.rst [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`3921`:, :ghissue:`3923`: Execute varoius tests in clean ``database_dir`` to avoid subsequent test flakeyness.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:216]: varoius -> various

File: src/docs/src/whatsnew/3.3.rst [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+* :ghissue:`4188`: Enable eunit coverage for all applications instead of enabling it per-application.
+
+* :ghissue:`4202`: Fix race condition in ddoc cache lru test 
+
+* :ghissue:`4203`, :ghissue:`4205`: Reduce test log noise.
+
+Docs
+----
+
+* :ghissue:`4146`: The CouchDB documentation has been moved into the main CouchDB repository.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:249]: #4146 -> #4164 Integrate docs in the main repo

File: src/docs/src/whatsnew/3.3.rst [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`4000`: Ensure ``Object.prototype.toSource()`` is always available
+
+* :ghissue:`4017`: Update ``mochiweb`` to 3.1.1.
+
+* :ghissue:`4018`: Update ``jiffy`` to 1.1.1 and ``b64url`` to 1.0.3.
+
+* :ghissue:`4021`: Reduce smoosh compaction log level to ``debug``.
+
+* :ghissue:`4135`: Allow and evaluate nested json claim roles in JWT token.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:71]: #4135 is: add ioq:call_search

#4041 Allow and evaluate nested json claim roles

File: src/docs/src/whatsnew/3.3.rst [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`3864`: Improve ``erlang_ls`` configuration.
+
+* :ghissue:`3853`: Remove legacy ``ddoc_cache_opener`` gen_server and speed up event routing.
+
+* :ghissue:`3879`: Remove use of ``ERL_OPTS`` environment variable. All supported Erlang version now use ``ERL_COMPILER_OPTIONS`` for the same purpose.
+
+* :ghissue:`3883`: Add support for SpiderMonkey 91.
+
+* :ghissue:`3889`: Track ``libicu`` collator versions in the view header
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:51]: missing `.` at the end

File: src/docs/src/whatsnew/3.3.rst [hunk 13]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`3927`: Include index signature in ``_search_info``.
+
+* :ghissue:`3963`: Optimtize key tree stemming by using maps instead of sets. This greatly reduced memory usage for heavily conflicted docs in some situations.
+
+* :ghissue:`3974`: Create new config options in ``[couchdb]`` and ``[smoosh]`` sections to enable finer control of compaction logging levels.
+
+* :ghissue:`3983`, :ghissue:`3984`, :ghissue:`3985`, :ghissue:`3987` :ghissue:`#4033`: Add various functions to ``couch_debug`` module.
+
+* :ghissue:`4000`: Ensure ``Object.prototype.toSource()`` is always available
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:63]: missing . at the end

File: src/docs/src/whatsnew/3.3.rst [hunk 14]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`4083`: Support Elixir 13.
+
+* :ghissue:`4085`: Add an option to let ``custodian`` always use ``[cluster] n`` value.
+
+* :ghissue:`4095`: Implement `winning_revs_only` option for the replicator. It replicates only the winning revisions from the source to the target, effectively discarding conflicts.
+
+* :ghissue:`4135`: Separate search IO from file IO.
+
+* :ghissue:`4140`, :ghissue:`4162`: Upgrade hash algorithm for cookie auth (sha1 -> sha256). This introduces a new config setting ``hash_algorithms``. New cookie values are hashed with sha256, sha1 hashes are still accepted. Admins can set this to sha256 only. Sha1 will be disallowed in the next major release. Show supported hash algorithms in ``_versions`` endpoint.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:87]: as above, would change to _versions end-point to

`/_node/_local/_versions`

File: src/docs/src/whatsnew/3.3.rst [hunk 15]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`4179`: Don't double-encode changes sequence strings in the replicator.
+
+* :ghissue:`4182`: Explicitly maintain a fully connected cluster. Previously, it was possible for the nodes to disconnect, and for that state to persist util the nodes restarted.
+
+* :ghissue:`4177`: Redact passwords in log file.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:93]: #4177 is: Remove the long deprecated bigcouch 0.4 change sequence support 

#4198 Redact passwords in log file

File: src/docs/src/whatsnew/3.3.rst [hunk 16]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`4050`: Handle ``all_dbs_active`` in ``fabric_doc_update``.
+
+* :ghissue:`4061`: Return a proper ``400`` error when an invlaid object is sent to ``_bulk_get``.
+
+* :ghissue:`4070`: Prevent error:function_clause in ``check_security/3`` if roles claim is malformed.
+
+* :ghissue:`4075`: Fix ``couch_debug:opened_files*`` functions.
+
+* :ghissue:`4108`: Trim `X-Auth-CouchDB-Roles` header after reading.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:185]:
```suggestion
* :ghissue:`4108`: Trim ``X-Auth-CouchDB-Roles`` header after reading.
```

File: src/docs/src/whatsnew/3.3.rst [hunk 17]
```diff
@@ -0,0 +1,263 @@
...
+
+* :ghissue:`4108`: Trim ``X-Auth-CouchDB-Roles`` header after reading.
+
+* :ghissue:`4153`: The ``require_valid_user`` setting is now under ``chttpd``.
+
+* :ghissue:`4161`: Fix ``content-type`` handling in ``_session``.
+
+* :ghissue:`4176`: Fix ``eventsource`` ``_changes`` feed.
+
+* :ghissue:`4197`: Support large (and impractical as-of-yet) ``q` values. Fix shard open timeouts for ``q > 64``.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:193]:
```suggestion
* :ghissue:`4197`: Support large (and impractical as-of-yet) ``q`` values. Fix shard open timeouts for ``q > 64``.
```

File: src/docs/src/whatsnew/3.3.rst [hunk 18]
```diff
@@ -0,0 +1,340 @@
...
+  ``database_dir`` to avoid subsequent test flakeyness.
+
+* :ghissue:`3968`: Ensure key tree rev stemming does’t take too much memory.
+
+* :ghissue:`3980`: Upgrade Mango test dependency ``nose`` to ``nose`` and fix
+  flakey-on-Windows tests.
+
+* :ghissue:`4006`: Remove CI support for Debian 9.
+
+* :ghissue:`4061`:, :ghissue:`4082`: Update PPC CI builder.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:296]:
```suggestion
* :ghissue:`4061`, :ghissue:`4082`: Update PPC CI builder.
```

File: src/docs/src/whatsnew/3.3.rst [hunk 19]
```diff
@@ -0,0 +1,340 @@
...
+
+* :ghissue:`4254`: The ``_dbs_info`` access control is now configured with the
+  ``[couchdb] admin_only_all_dbs`` setting. Defaults to true. This was a
+  leftover from the 3.0.0 release.
+
+Performance
+-----------
+
+* :ghissue:`3860`: Add sharding to ``couch_index_server``, similar to
+  :ghissue:`3366`:, avoids processing bottlenecs on servers with a lot of
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:133]:
```suggestion
  :ghissue:`3366`, avoids processing bottlenecs on servers with a lot of
```

File: src/docs/src/whatsnew/3.3.rst [hunk 20]
```diff
@@ -0,0 +1,340 @@
...
+* :ghissue:`3927`: Include index signature in ``_search_info``.
+
+* :ghissue:`3963`: Optimtize key tree stemming by using maps instead of
+  sets. This greatly reduced memory usage for heavily conflicted docs in some
+  situations.
+
+* :ghissue:`3974`: Create new config options in ``[couchdb]`` and ``[smoosh]``
+  sections to enable finer control of compaction logging levels.
+
+* :ghissue:`3983`, :ghissue:`3984`, :ghissue:`3985`, :ghissue:`3987`
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:77]:
```suggestion
* :ghissue:`3983`, :ghissue:`3984`, :ghissue:`3985`, :ghissue:`3987`,
```

File: src/docs/src/whatsnew/3.3.rst [hunk 21]
```diff
@@ -0,0 +1,340 @@
...
+* :ghissue:`4069`, :ghissue:`4084`: Drop support for Erlang < 23, update
+  ``vm.args`` settings to match. Review this if you have customized your
+  ``vm.args``.
+
+* :ghissue:`4083`: Support Elixir 13.
+
+* :ghissue:`4085`: Add an option to let ``custodian`` always use ``[cluster] n``
+  value.
+
+* :ghissue:`4095`: Implement `winning_revs_only` option for the replicator. It
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:102]:
```suggestion
* :ghissue:`4095`: Implement ``winning_revs_only`` option for the replicator. It
```

File: src/docs/src/whatsnew/3.3.rst [hunk 22]
```diff
@@ -0,0 +1,340 @@
...
+
+* :ghissue:`3860`: Add sharding to ``couch_index_server``, similar to
+  :ghissue:`3366`:, avoids processing bottlenecs on servers with a lot of
+  concurrent view indexing going on.
+
+* :ghissue:`3891`: Avoid decoding JWT payloads when not necessary.
+
+* :ghissue:`4031`: Default ``[rexi] use_kill_all`` to ``true``. This improves
+  intra-cluster-node messaging. Set to false if you run a cluster with nodes
+  tthat have a version <3.0.0.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:140]:
```suggestion
  that have a version <3.0.0.
```

File: src/docs/src/whatsnew/3.3.rst [hunk 23]
```diff
@@ -0,0 +1,340 @@
...
+
+* :ghissue:`4040`: Handle ``exit(shutdown)`` error in ``chttpd``.
+
+* :ghissue:`4043`: Fix purge request timeouts (5s -> infinity).
+
+* :ghissue:`4146`: The ``devcontainer`` has been updated.
+
+* :ghissue:`4050`: Handle ``all_dbs_active`` in ``fabric_doc_update``.
+
+* :ghissue:`4160`: Return a proper ``400`` error when an invlaid object is sent
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:241]:
```suggestion
* :ghissue:`4160`: Return a proper ``400`` error when an invalid object is sent
```

File: src/docs/src/whatsnew/3.3.rst [hunk 24]
```diff
@@ -0,0 +1,340 @@
...
+  multicore server.
+
+Tests
+-----
+
+* :ghissue:`3825`: Eliminate Elixir compiler warnings.
+
+* :ghissue:`3830`: Reduce skipped Elixir integration tests.
+
+* :ghissue:`3890`: Handle ``not_found`` lookups removing ddoc cache key
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:275]:
```suggestion
* :ghissue:`3890`: Handle ``not_found`` lookups removing ddoc cache key.
```

File: src/docs/src/whatsnew/3.3.rst [hunk 25]
```diff
@@ -0,0 +1,340 @@
...
+* :ghissue:`4132`: Address race condition in cpse_incref_decref test.
+
+* :ghissue:`4151`: Refactor replication tests to use clustered endpoints.
+
+* :ghissue:`4178`: Add test coverage to prevent junk in ``eventsource``.
+
+* :ghissue:`4188`: Enable eunit coverage for all applications instead of
+  enabling it per-application.
+
+* :ghissue:`4202`: Fix race condition in ddoc cache lru test
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:316]:
```suggestion
* :ghissue:`4202`: Fix race condition in ddoc cache lru test.
```

File: src/docs/src/whatsnew/3.3.rst [hunk 26]
```diff
@@ -0,0 +1,340 @@
...
+
+Docs
+----
+
+* :ghissue:`4164`: The CouchDB documentation has been moved into the main
+  CouchDB repository.
+
+* :ghissue:`4174`: Update Sphinx to version 5.1.1
+
+* :ghissue:`4170`: Document the ``_versions`` endpoint.
```
Comment (big-r81) [inline src/docs/src/whatsnew/3.3.rst:328]:
```suggestion
* :ghissue:`4170`: Document the ``/_node/_local/_versions`` endpoint.
```

File: src/docs/src/whatsnew/3.3.rst [hunk 27]
```diff
@@ -0,0 +1,371 @@
...
+
+Features and Enhancements
+-------------------------
+
+* :ghissue:`3766`, :ghissue:`3970`, :ghissue:`3972`, :ghissue:`4093`,
+  :ghissue:`4102`, :ghissue:`4104`, :ghissue:`4110`, :ghissue:`4111`,
+  :ghissue:`4114`, :ghissue:`4245`, :ghissue:`4246`:, :ghissue:`4266`: Add
+  ``smoosh`` queue persistence. This allows resuming ``smoosh`` operations
+  after a node restart. This is disabled by default and can be enabled with
+  ``[smoosh] persist = true``. Optimise ``smoosh`` operationsand increase
```
Comment (nickva) [inline src/docs/src/whatsnew/3.3.rst:36]: missing space in `operationsand`?

File: src/docs/src/whatsnew/3.3.rst [hunk 28]
```diff
@@ -0,0 +1,371 @@
...
+* :ghissue:`3883`: Add support for SpiderMonkey 91.
+
+* :ghissue:`3889`: Track ``libicu`` collator versions in the view header.
+
+* :ghissue:`3952`: Make the timeout for receiving requests from attachment
+  writers configurable.
+
+* :ghissue:`3927`: Include index signature in ``_search_info``.
+
+* :ghissue:`3963`: Optimtize key tree stemming by using maps instead of
```
Comment (nickva) [inline src/docs/src/whatsnew/3.3.rst:68]: `Optimtize` -> `Optimise`

File: src/docs/src/whatsnew/3.3.rst [hunk 29]
```diff
@@ -0,0 +1,371 @@
...
+* :ghissue:`4018`: Update ``jiffy`` to 1.1.1 and ``b64url`` to 1.0.3.
+
+* :ghissue:`4021`: Reduce smoosh compaction log level to ``debug``.
+
+* :ghissue:`4041`: Allow and evaluate nested json claim roles in JWT token.
+
+* :ghissue:`4060`, :ghissue:`4290`: Add support for Erlang 25.
+
+* :ghissue:`4064`: Enable replicating purge requests between nodes. Also avoid
+  apploying interactive purges more than once.
```
Comment (nickva) [inline src/docs/src/whatsnew/3.3.rst:89]: `apploying` -> `applying`

File: src/docs/src/whatsnew/3.3.rst [hunk 30]
```diff
@@ -0,0 +1,371 @@
...
+
+* :ghissue:`3898`: Remove CI support for Ubuntu 16.04.
+
+* :ghissue:`3903`, :ghissue:`3914`: Refactor Jenkins to dynamically generate
+  stages. Drop ``MINIMUM_ERLANG_VERSION`` to 20, drop the packaging
+  ``ERLANG_VERSION`` to 23, add the ``weatherreport-test`` as a build step, and
+  add ``ARM`` and ``POWER`` back into the matrix.
+
+* :ghissue:`3921`:, :ghissue:`3923`: Execute various tests in clean
+  ``database_dir`` to avoid subsequent test flakeyness.
```
Comment (nickva) [inline src/docs/src/whatsnew/3.3.rst:316]: `flakeyness` vs `flakiness`, though both might work?

File: src/docs/src/whatsnew/3.3.rst [hunk 31]
```diff
@@ -0,0 +1,371 @@
...
+
+* :ghissue:`3968`: Ensure key tree rev stemming does’t take too much memory.
+
+* :ghissue:`3980`: Upgrade Mango test dependency ``nose`` to ``nose`` and fix
+  flakey-on-Windows tests.
```
Comment (nickva) [inline src/docs/src/whatsnew/3.3.rst:321]: `flakey` -> `flaky` ?

File: src/docs/src/whatsnew/3.3.rst [hunk 32]
```diff
@@ -0,0 +1,371 @@
...
+
+* :ghissue:`4006`: Remove CI support for Debian 9.
+
+* :ghissue:`4061`, :ghissue:`4082`: Update PPC CI builder.
+
+* :ghissue:`4096`: Fix flakey ``valudate_doc_update`` Elixir test.
```
Comment (nickva) [inline src/docs/src/whatsnew/3.3.rst:327]: `flakey` -> `flaky`

Comment (nickva) [inline src/docs/src/whatsnew/3.3.rst:327]: `valudate` -> `validate`

File: src/docs/src/whatsnew/3.3.rst [hunk 33]
```diff
@@ -0,0 +1,371 @@
...
+* :ghissue:`4132`: Address race condition in ``cpse_incref_decref`` test.
+
+* :ghissue:`4151`: Refactor replication tests to use clustered endpoints.
+
+* :ghissue:`4178`: Add test coverage to prevent junk in ``eventsource``.
+
+* :ghissue:`4188`: Enable eunit coverage for all applications instead of
+  enabling it per-application.
+
+* :ghissue:`4202`: Fix race condition in ddoc cache lru test.
```
Comment (nickva) [inline src/docs/src/whatsnew/3.3.rst:345]: `lru` -> `LRU`?

**Review by nickva — DISMISSED** [review]
A few tiny spelling nits

**Review by nickva — APPROVED** [review]
+1

**Pull request merged**